### PR TITLE
PLT-5847 marlowe-contract service

### PR DESCRIPTION
**Pull request by jhbertra**
- [x] Adds new `marlowe-contract` service to runtime topology
- [x] Adds `MarloweLoad` protocol to `MarloweRuntime` protocol
- [x] Adds `load` command to `marlowe-runtime-cli`
- [x] Updates docker-compose, operable and OCI images
- [x] TODO Updates nomad tasks
- [x] TODO adds documentation for JSON format accepted by `load` command.

Pre-submit checklist:
- Branch
    - [x] Tests are provided (if possible)
    - [x] Commit sequence broadly makes sense
    - [x] Key commits have useful messages
    - [x] Formatting, PNG optimization, etc. are updated
- PR
    - [x] Self-reviewed the diff
    - [x] Useful pull request description
        - Review required
        - [x] Substantial changes to code, test, or documentation
    - [x] Reviewer requested

### CODE REVIEW

File: marlowe-runtime-cli/app/Language/Marlowe/Runtime/CLI/Command/Load.hs [hunk 1]
```diff
@@ -0,0 +1,333 @@
...
+  }
+
+loadCommandParser :: ParserInfo LoadCommand
+loadCommandParser = info parser $ progDesc "Load a contract into the runtime"
+  where
+    parser = LoadCommand
+      <$> contractFileOption
+    contractFileOption = strArgument $ mconcat
+      [ metavar "FILE_PATH"
+      , help "A file that contains the JSON representation of the contract to load."
```
Comment (jhbertra) [inline marlowe-runtime-cli/app/Language/Marlowe/Runtime/CLI/Command/Load.hs:54]: TODO document this better

File: marlowe-runtime-cli/app/Language/Marlowe/Runtime/CLI/Command/Load.hs [hunk 2]
```diff
@@ -0,0 +1,333 @@
...
+  Assert _ c -> countNodes' (count + 1) c
+  Import path -> countNodes count path
+
+countCase :: Int -> LoadCase -> CountM Int
+countCase count (LoadCase _ c) = countNodes' (count + 1) c
+
+
+withContract :: FilePath -> (LoadContract -> CountM a) -> CountM a
+withContract path m = do
+  fileExists <- doesFileExist path
```
Comment (jhbertra) [inline marlowe-runtime-cli/app/Language/Marlowe/Runtime/CLI/Command/Load.hs:123]: TODO find a way to unify this duplicated code.

File: deploy/operables.nix [hunk 3]
```diff
@@ -384,6 +384,38 @@ in
...
+      # REQUIRED VARS #
+      #################
+      # HOST, PORT: network binding
+      # STORE_DIR: location of the contract store directory
+      # HTTP_PORT: port number for the HTTP healthcheck server
+
+      #################
+      # OPTIONAL VARS #
+      #################
+      OTEL_EXPORTER_OTLP_ENDPOINT: The url of the open telemetry collector
```
Comment (bwbush) [inline deploy/operables.nix:401]:
```suggestion
      # OTEL_EXPORTER_OTLP_ENDPOINT: The url of the open telemetry collector
```

File: marlowe-runtime-cli/app/Language/Marlowe/Runtime/CLI/Command/Load.hs [hunk 4]
```diff
@@ -0,0 +1,322 @@
...
+    Right hash -> do
+      hPutStrLn stderr "\nDone."
+      putStrLn $ read $ show hash
+
+type CountM = ReaderT [FilePath] (ExceptT LoadError CLI)
+
+countNodes :: Int -> FilePath -> CountM Int
+countNodes count path = withContract path $ countNodes' count
+
+countNodes' :: Int -> LoadContract -> CountM Int
```
Comment (bwbush) [inline marlowe-runtime-cli/app/Language/Marlowe/Runtime/CLI/Command/Load.hs:94]: I wonder if the multiplate in `Language.Marlowe.Core.V1.Plate` could be used to do this counting?

Comment (jhbertra) [reply marlowe-runtime-cli/app/Language/Marlowe/Runtime/CLI/Command/Load.hs:94]: Possibly, the concept of a `Node` is somewhat abstract though, and isn't directly captured by anything in there.

Comment (bwbush) [reply marlowe-runtime-cli/app/Language/Marlowe/Runtime/CLI/Command/Load.hs:94]: Actually, I see that `LoadContract` is not `Contract`, which is what the plate acts on. :frowning_face:

File: marlowe-runtime/contract/Language/Marlowe/Runtime/Contract/Store/File.hs [hunk 5]
```diff
@@ -0,0 +1,240 @@
...
+import Language.Marlowe.Runtime.Cardano.Api (fromCardanoDatumHash, toCardanoScriptData)
+import Language.Marlowe.Runtime.ChainSync.Api (DatumHash(DatumHash), toDatum)
+import Language.Marlowe.Runtime.Contract.Store
+import Language.Marlowe.Runtime.Core.Api ()
+import Plutus.V2.Ledger.Api (fromBuiltin)
+import System.FilePath (takeBaseName, (<.>), (</>))
+import System.IO.LockFile (withLockFile)
+import UnliftIO
+import UnliftIO.Directory
+  ( XdgDirectory(XdgData)
```
Comment (bwbush) [inline marlowe-runtime/contract/Language/Marlowe/Runtime/Contract/Store/File.hs:36]: Is the XDG stuff portable?

File: marlowe-runtime/marlowe-contract/Main.hs [hunk 6]
```diff
@@ -0,0 +1,213 @@
...
+      , value 3727
+      , metavar "PORT_NUMBER"
+      , help "The port number to run the marlowe load server on."
+      , showDefault
+      ]
+
+    bufferSizeParser = option readOption $ mconcat
+      [ long "buffer-size"
+      , short 'b'
+      , value 4098
```
Comment (bwbush) [inline marlowe-runtime/marlowe-contract/Main.hs:172]: Seems too high. Are exceptions caught in a way the forces a flush before exit?

**Review by bwbush — APPROVED** [review]
Looks great. I'll do an on-chain test when `marlowe-runtime-cli create` supports this.

File: marlowe-runtime/contract/Language/Marlowe/Runtime/Contract/Store/File.hs (hunk 5 above)
Comment (jhbertra) [reply marlowe-runtime/contract/Language/Marlowe/Runtime/Contract/Store/File.hs:36]: Yes it is

File: marlowe-runtime/marlowe-contract/Main.hs (hunk 6 above)
Comment (jhbertra) [reply marlowe-runtime/marlowe-contract/Main.hs:172]: 1024 could be a better default. The staging area is wrapped in a bracket that discards the entire temp staging directory on exception. So there should be no resource leak (though nothing will protect from a SIGKILL or a loss of power of course).

**Pull request merged**